pkg/controller/storagecluster/cephcluster.go

```diff
@@ -564,3 +575,43 @@ func (r *ReconcileStorageCluster) checkTuneStorageDevices(ds ocsv1.StorageDevice
 	// not a known disk type, don't tune
 	return diskSpeedUnknown, nil
 }
+
+func generateStretchClusterSpec(sc *ocsv1.StorageCluster) *cephv1.StretchClusterSpec {
+
+	var zones []string
+	stretchClusterSpec := cephv1.StretchClusterSpec{}
+	stretchClusterSpec.FailureDomainLabel, zones = sc.Status.NodeTopologies.GetKeyValues(determineFailureDomain(sc))
+
+	for _, zone := range zones {
+		if zone == sc.Spec.NodeTopologies.ArbiterLocation {
+			continue
+		}
+		stretchClusterSpec.Zones = append(stretchClusterSpec.Zones, cephv1.StretchClusterZoneSpec{
+			Name:    zone,
+			Arbiter: false,
+		})
+	}
+	stretchClusterSpec.Zones = append(stretchClusterSpec.Zones, cephv1.StretchClusterZoneSpec{
+		Name:    sc.Spec.NodeTopologies.ArbiterLocation,
+		Arbiter: true,
+	})
+
+	return &stretchClusterSpec
```

Comment:
```suggestion
	return stretchClusterSpec
```

Reply:
cephv1.MonSpec requires *cephv1.StretchClusterSpec. It just makes it easy to match with what Cephv1 defines it as.

Reply:
I think this is fine in this case.